test/new-e2e/tests/agent-platform/common/agent_behaviour.go

```diff
@@ -314,22 +314,36 @@ func CheckCWSBehaviour(t *testing.T, client *TestClient) {
 	t.Run("system-probe and security-agent communicate", func(tt *testing.T) {
 		var statusOutputJSON map[string]any
 		var result bool
-		for try := 0; try < 10 && !result; try++ {
+		for try := 1; try <= 20 && !result; try++ {
 			status, err := client.Host.Execute("sudo /opt/datadog-agent/embedded/bin/security-agent status -j")
-			if err == nil {
-				statusLines := strings.Split(status, "\n")
-				status = strings.Join(statusLines[1:], "\n")
-				err := json.Unmarshal([]byte(status), &statusOutputJSON)
-				require.NoError(tt, err)
-				if runtimeStatus, ok := statusOutputJSON["runtimeSecurityStatus"]; ok {
-					if connected, ok := runtimeStatus.(map[string]any)["connected"]; ok {
-						result = connected == true
-					}
-				}
+			if err != nil {
+				tt.Logf("[CheckCWSBehaviour] Try #%d: unable to retrieve security-agent status with error %s", try, err)
+				time.Sleep(1 * time.Second)
+				continue
+			}
+
+			statusLines := strings.Split(status, "\n")
+			status = strings.Join(statusLines[1:], "\n")
+			err = json.Unmarshal([]byte(status), &statusOutputJSON)
+			require.NoError(tt, err)
+
+			runtimeStatus, ok := statusOutputJSON["runtimeSecurityStatus"]
+			require.True(tt, ok, "runtimeSecurityStatus should be present on the security-agent status")
+
+			connected, ok := runtimeStatus.(map[string]any)["connected"]
+			require.True(tt, ok, "connected should be present on the runtimeSecurityStatus")
+
+			result, ok := connected.(bool)
```

Comment:
The tests are failing, I think we might be shadowing the `result` from the outer scope here.
```suggestion
			result, ok = connected.(bool)
```

Reply:
It wont work without declaring the `ok` as well. let me address it.

Reply:
Declaring `ok` is not needed because it's already declared above (on line 330) at least; declaring it above is making the linter fail.